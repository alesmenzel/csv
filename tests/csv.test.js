const fs = require('fs')
const path = require('path')
const {pipeline} = require('stream')
const { promisify } = require('util')
const Parser = require('../csv-parser')
const Collect = require('./collect')

const pipelineAsync = promisify(pipeline)
const files = fs.readdirSync(path.resolve(__dirname, '../csv')).map(file => [file])

it.each(files)('CSV: %s', async (file) => {
  const parser = new Parser()
  const collect = new Collect()
  const input = fs.createReadStream(path.resolve(__dirname, '../csv/', file))
  await pipelineAsync(input, parser, collect)
  expect(collect.getData()).toMatchSnapshot(file)
})
