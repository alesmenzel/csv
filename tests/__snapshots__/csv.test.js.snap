// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV: empty-cells.csv: empty-cells.csv 1`] = `
Array [
  Array [
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
  ],
  Array [
    "",
    "",
    "",
    "",
    "",
  ],
]
`;

exports[`CSV: empty-file.csv: empty-file.csv 1`] = `Array []`;

exports[`CSV: json-data.csv: json-data.csv 1`] = `
Array [
  Array [
    "A",
    "B",
    "C",
  ],
  Array [
    "a",
    "_-_-_-[{\\"title\\":\\"aaaaaa\\",\\"abstract\\":\\"\\",\\"article_id\\":\\"82ec82fb-33b2-42e8-a8aa-2f19cc7a135e\\",\\"widgets\\":\\"[{\\\\\\"type\\\\\\":\\\\\\"Img\\\\\\",\\\\\\"options\\\\\\":{\\\\\\"singleImg\\\\\\":{\\\\\\"media_id\\\\\\":\\\\\\"af939acc-a252-4712-9f51-72afef0fac4a.jpg\\\\\\",\\\\\\"src\\\\\\":\\\\\\"http://google.com\\\\\\",\\\\\\"type\\\\\\":0,\\\\\\"star\\\\\\"",
    "b",
  ],
  Array [
    "a",
    "{\\"a\\":3,\\"b\\":4,\\"c\\":{\\"some\\":\\"string\\"}}",
    "c",
  ],
]
`;

exports[`CSV: kitchen-sink.csv: kitchen-sink.csv 1`] = `
Array [
  Array [
    "A",
    "B",
    "C",
  ],
  Array [
    "",
    "",
    "",
  ],
  Array [
    "a\\"",
    "b",
    "",
  ],
  Array [
    "a\\\\\\"",
    "b",
    "c",
  ],
  Array [
    "a

",
    "b

",
    "c",
  ],
  Array [
    "a",
    "{\\"a\\":3,\\"b\\":4,\\"c\\":{\\"some\\":\\"string\\"}}",
    "c",
  ],
  Array [
    "\\\\x\\"\\"x\\"\\"\\"",
    "\\\\y",
    "z\\\\\\\\\\\\\\\\\\\\\\"",
  ],
  Array [
    "a",
    "b,",
    "c",
  ],
]
`;

exports[`CSV: newlines.csv: newlines.csv 1`] = `
Array [
  Array [
    "A",
    "B",
    "C",
  ],
  Array [
    "a

",
    "b

",
    "c",
  ],
  Array [
    "x",
    "y",
    "z

",
  ],
]
`;

exports[`CSV: quoted.csv: quoted.csv 1`] = `
Array [
  Array [
    "A",
    "B",
    "C",
  ],
  Array [
    "a",
    "b",
    "c",
  ],
  Array [
    "x",
    "y",
    "z",
  ],
]
`;

exports[`CSV: quoted-with-escape.csv: quoted-with-escape.csv 1`] = `
Array [
  Array [
    "A\\\\\\"",
    "B",
    "C",
  ],
  Array [
    "\\\\a",
    "b\\\\\\\\\\\\\\"",
    "c",
  ],
  Array [
    "\\\\x",
    "\\\\y",
    "z\\\\\\\\\\\\\\\\\\\\\\"",
  ],
]
`;

exports[`CSV: quoted-with-escape-and-quote.csv: quoted-with-escape-and-quote.csv 1`] = `
Array [
  Array [
    "A\\\\\\"\\"",
    "B",
    "C",
  ],
  Array [
    "\\\\a\\"",
    "b\\\\\\\\\\"\\\\\\"",
    "\\"\\"\\"\\"",
  ],
  Array [
    "\\\\x\\"\\"x\\"\\"\\"",
    "\\\\y",
    "z\\\\\\\\\\\\\\\\\\\\\\"",
  ],
]
`;

exports[`CSV: quoted-with-quotes.csv: quoted-with-quotes.csv 1`] = `
Array [
  Array [
    "A\\"",
    "B",
    "C",
  ],
  Array [
    "a",
    "b\\"",
    "c",
  ],
  Array [
    "x",
    "y",
    "z\\"",
  ],
]
`;

exports[`CSV: separator-in-data.csv: separator-in-data.csv 1`] = `
Array [
  Array [
    "A",
    ",B,,",
    "C",
  ],
  Array [
    "a",
    "b,",
    "c",
  ],
  Array [
    "x",
    ",y",
    "z",
  ],
]
`;

exports[`CSV: simple.csv: simple.csv 1`] = `
Array [
  Array [
    "A",
    "B",
    "C",
  ],
  Array [
    "a",
    "b",
    "c",
  ],
  Array [
    "x",
    "y",
    "z",
  ],
]
`;

exports[`CSV: simple-quoted-bom.csv: simple-quoted-bom.csv 1`] = `
Array [
  Array [
    "A",
    "B",
    "C",
  ],
  Array [
    "a",
    "b",
    "c",
  ],
  Array [
    "x",
    "y",
    "z",
  ],
]
`;
